ethcore/src/client/client.rs

```diff
@@ -2756,9 +2758,12 @@ impl IoChannelQueue {
 	{
 		let queue_size = self.currently_queued.load(AtomicOrdering::Relaxed);
 		if queue_size >= self.limit {
-			return Err(EthcoreError::FullQueue(self.limit))
+			let err_limit = usize::try_from(self.limit).unwrap_or(usize::max_value());
+			return Err(EthcoreError::FullQueue(err_limit))
 		};
 
+		let count = i64::try_from(count).unwrap_or(i64::max_value());
+
 		let currently_queued = self.currently_queued.clone();
 		let _ok = channel.send(ClientIoMessage::execute(move |client| {
```

Comment:
Probably a silly question but why not check for an error here and only increment if the `send()` was `Ok`?

```
let send_result =  channel.send(ClientIoMessage::execute(move |client| {
    …
}));
match send_result {
    Ok(()) => self.currently_queued.fetch_add(count, AtomicOrdering::SeqCst),
    Err(e) => return Err(EthcoreError::Io(e)),
}
Ok(())
```

Reply:
That is how the code looked like a couple of months back, it was changed to avoid that match statement (not by me).
But it essentially does the same as the current implementation, just in a slightly different style, has no effect on the behavior.